task/sast-find-unicode-control/0.1/sast-find-unicode-control.yaml

```diff
@@ -0,0 +1,232 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: sast-find-unicode-control
+spec:
+  description: >-
+    Scans source code for non-printable unicode characters in all text files.
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - description: Image URL.
+      name: image-url
+      type: string
+      # In a future 0.4 version of the task, drop the default to make this required
+      default: ""
+    - description: Image digest to report findings for.
+      name: image-digest
+      type: string
+      # In a future 0.4 version of the task, drop the default to make this required
+      default: ""
+    - name: FIND_UNICODE_CONTROL_GIT_URL
+      type: string
+      description: URL from repository to find unicode control
+      default: "https://github.com/siddhesh/find-unicode-control.git"
```

Comment:
If the task runs are expected to be reproducible, we need to pin the scanner to specific revision.  This expectation is unmet with the existing `sast-snyk-check` task because it uses an external scanning service, which can be updated anytime.  Nevertheless, given the fact that we are cloning a personal git repository rather than an official project with a well-defined release procedure,  I would say we should pin the scanner to a specific (verified) revision.

@14rcole What is the preference of the Konflux team on this?

@hanchuntao We should also record the revision of the used scanner in scan properties via `csgrep --set-scan-prop` in a similar way as we record the revision of KFP: https://github.com/csutils/csdiff/pull/207/files

Reply:
Updated, the final result is:
```
"scan": {
[sast-unicode-check]         "find-unicode-control": "https://github.com/siddhesh/find-unicode-control.git c2accbfbba7553a8bc1ebd97089ae08ad8347e58",
[sast-unicode-check]         "known-false-positives-git-url": "https://gitlab.cee.redhat.com/osh/known-false-positives.git#297df9a6d1a62e0c16a974e8cf1f3770a3b0dc4c"
[sast-unicode-check]     },
```